petl/io/json.py

```diff
@@ -173,6 +171,22 @@ def __iter__(self):
         return iterdicts(self.dicts, self.header, self.sample, self.missing)
 
 
+def iterjlines(f, header, missing):
+    it = iter(f)
+
+    if header is None:
+        header = list()
+        peek, it = iterpeek(it, 1)
```

Comment:
JSON lines file will have complete schema on first line or any other line, so we can safely peek just one line.